src/theme/components/button.ts

```diff
@@ -2,10 +2,20 @@ import { defineStyleConfig } from "@chakra-ui/react";
 
 export const buttonTheme = defineStyleConfig({
   baseStyle: {
+    border: "1px solid rgba(255, 255, 255, 0)",
     borderRadius: "base",
     px: 6,
     py: 3,
     whiteSpace: "normal",
+    display: "inline-block",
+    _before: {
```

Comment:
Why this change for the width/height hover fix?

Reply:
The issue of the width change was caused by the change in `fontWeight` on hover. 

The size of the button is determined by the horizontal size of the text.  We can't adjust by a predetermined amount, because we don't know how much changing the `fontWeight` will change the size of the button.

To correct this, I used the ::before pseudo-element to insert an invisible copy of the button's text with the same fontWeight as the hover, which increases the base width to that of the hover state.